app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -49,154 +54,324 @@ class PipelineViz extends React.Component {
+      const { from, to } = edgeData;
+      from && this.getStepDataAt(from).outputEdges.push(edgeIndex);
+      to && this.getStepDataAt(to).inputEdges.push(edgeIndex);
+    });
+  }
+
+  getEdgesBetweenNodes(graph, fromId, toId, type) {
+    const allEdges = graph.getEdgesBetweenNodes(fromId, toId);
+    if (type) {
+      const edgeTypeRegex = new RegExp(`-${type}$`, "g");
+      return allEdges.filter(edgeId => edgeId.match(edgeTypeRegex));
+    }
+    return allEdges;
+  }
+
+  getModifiedStepName(stepClassName) {
+    return stepClassName.replace(/^(PipelineStep(Run|Generate)?)/, "");
+  }
 
   handleStepClick(stageIndex, info) {
     const clickedNodeId = info.nodes[0];
     if (clickedNodeId == null) {
       return;
     }
+    this.graphs.forEach((graph, i) => i != stageIndex && graph.unselectAll());
 
-    const stepData = this.getStepDataAtIndices(stageIndex, clickedNodeId);
-    const inputFiles = stepData.inputInfo.map(input => {
-      const fileInfo = {
-        fileName: input.fileName,
-        url: input.url,
+    const inputEdgesInfo = this.getEdgeInfoFor(
+      stageIndex,
+      clickedNodeId,
+      "input"
+    );
+    const inputInfo = inputEdgesInfo.map(edgeInfo => {
+      const fromStepName = edgeInfo.from
+        ? this.getStepDataAt(edgeInfo.from).name
+        : "";
+      return {
+        fromStepName: fromStepName,
+        files: edgeInfo.files,
       };
-      if (input.fromStageIndex != null && input.fromStepIndex != null) {
-        fileInfo.fromStepName = this.getStepDataAtIndices(
-          input.fromStageIndex,
-          input.fromStepIndex
-        ).name;
-      } else {
-        fileInfo.fromStepName = "";
-      }
-      return fileInfo;
     });
 
+    const seenFiles = new Set();
+    const outputEdgesInfo = this.getEdgeInfoFor(
+      stageIndex,
+      clickedNodeId,
+      "output"
+    );
+    const outputInfo = outputEdgesInfo
+      .map(edgeInfo => {
+        // Remove duplicate output file listings
+        return edgeInfo.files.reduce((files, fileInfo) => {
+          if (!seenFiles.has(fileInfo.fileName)) {
+            seenFiles.add(fileInfo.fileName);
+            files.push(fileInfo);
+          }
+          return files;
+        }, []);
+      })
+      .flat();
+
+    const stepName = this.getStepDataAt({
+      stageIndex: stageIndex,
+      stepIndex: clickedNodeId,
+    }).name;
     this.setState({
       sidebarVisible: true,
       sidebarParams: {
-        stepName: stepData.name,
+        stepName: stepName,
         description: "",
-        inputFiles: inputFiles,
-        outputFiles: stepData.outputInfo,
+        inputFiles: inputInfo,
+        outputFiles: outputInfo,
       },
     });
   }
 
+  handleNodeHover(stageIndex, info) {
+    const { highlightColor } = this.props;
+    const graph = this.graphs[stageIndex];
+    const updatedInterStageArrows = [...this.state.interStageArrows];
+
+    const inputColorOptions = {
```

Comment:
Would it make sense for this to be a prop or constant?

Reply:
I set the #000000 to a prop, but the entire object I left as a const since the format of it is pretty specific to visjs

Reply:
Sorry, I meant the entire object being a constant outside this function. Since it did not change it would be unnecessary to redefine it on the function every time...

Reply:
I think I'm going to leave it as a prop just to stay consistent with how the other edge color updates are written!